fuel-merkle/src/binary/in_memory.rs

```diff
@@ -62,6 +67,54 @@ impl Default for MerkleTree {
     }
 }
 
+#[derive(Default, Debug, Clone)]
+pub struct MerkleRootCalculator {
+    stack: Vec<Node>,
+}
+
+impl MerkleRootCalculator {
+    pub fn new() -> Self {
+        Self { stack: Vec::new() }
+    }
+
+    pub fn push(&mut self, data: &[u8]) {
+        let node = Node::create_leaf(0, data);
```

Comment:
I am a little surprised that this works without a specifying the correct leaf index. But if it works, it works!